src/app/services/debug.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { DebugNode } from '../models/DebugNode';
+import { ApiService } from './api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DebugService {
+  private debugNode: Subject<any> = new Subject<any>();
+
+  constructor(private api: ApiService) { }
+
+  node$ = this.debugNode.asObservable();
```

Comment:
Why is it needed?

Reply:
This is needed to avoid multiple requests to the server, from multiple components.
Only the layout component retrieves the debug details information, and then passes it to the child components via this observable.
Otherwise I would have had to do multiple requests, to retrieve the debug details from the API.